pydantic/fields.py

```diff
@@ -594,9 +594,11 @@ def get_default(self, *, call_default_factory: bool = False, validated_data: dic
         if self.default_factory is None:
             return _utils.smart_deepcopy(self.default)
         elif call_default_factory:
-            if validated_data is None:
-                raise ValueError("'validated_data' must be provided if 'call_default_factory' is True.")
             if _fields.takes_validated_data_argument(self.default_factory):
+                if validated_data is None:
+                    raise ValueError(
+                        "'validated_data' is missing but should be provided. This is a bug, please report this."
```

Comment:
```suggestion
                       "The default factory requires the 'validated_data' argument, which was not provided when calling 'get_default'."
```

I don't think mentioning "this is a bug" is correct, as users could really just call `get_default` in the wrong way.